src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -21,17 +21,17 @@ internal sealed class AggregationManager
         // these fields are modified after construction and accessed on multiple threads, use lock(this) to ensure the data
         // is synchronized
         private readonly List<Predicate<Instrument>> _instrumentConfigFuncs = new();
-        private TimeSpan _collectionPeriod;
+        public TimeSpan CollectionPeriod { get; private set; }
```

Comment:
Nit: consider scoping the lock to this property, rather than 'this' + _aggregationManager